plugins/skipmap/index.js

```diff
@@ -0,0 +1,158 @@
+      required: false,
+      description: 'How long the vote should go.',
+      default: 5 * 60 * 1000
+    },
+
+    startTimer: {
+      required: false,
+      description: 'Time to wait before allowing the vote to go through.',
+      default: 15 * 60 * 1000
+    },
+
+    pastVoteTimer: {
+      required: false,
+      description: 'Time that needs to have passed since the last vote.',
+      default: 10 * 60 * 1000
+    },
+
+    endTimer: {
+      required: false,
+      description: 'Time before the end timer of the round where the votes are no longer valid.',
+      default: 15 * 60 * 1000
+    },
+
+    minimumVotes: {
+      required: false,
+      description: 'The minimum required amount of votes for the vote to go through',
+      default: 20
+    },
+
+    reminderInterval: {
+      required: false,
+      description: 'The time between individual reminders.',
+      default: 2 * 60 * 1000
+    }
+  },
+
+  init: async (server, options) => {
+    let votePos = 0;
+    let voteNeg = 0;
+    let intervalReminderBroadcasts;
+    let timeoutVote;
+    let timeLastVote;
+    let voteActive;
+    const playerVotes = {};
+
+    server.on(CHAT_MESSAGE, (info) => {
+      // Run through conditions
+      // check if message is command
+      if (!info.message.chat.startsWith(options.command)) return;
+      // check if enough time has passed since start of round and if not, inform the player
+      if (server.layerHistory[0].time < options.startTimer) {
+        const min = Math.floor(
+          ((options.startTimer - server.layerHistory[0].time) % (1000 * 60 * 60)) / (1000 * 60)
+        );
+        const sec = Math.floor(
+          (options.startTimer - server.layerHistory[0].time) % ((1000 * 60) / 1000)
+        );
+        server.rcon.warn(
+          info.steamID,
+          'Not enough time has passed since the start of the match. Please try again in ' +
+            min +
+            'm ' +
+            sec +
+            's'
+        );
+        return;
+      }
+      // check if enough time remains in the round, if not, inform player
+      if (server.layerHistory[0].time < 2 * 60 * 60 * 1000 - options.endTimer) {
+        server.rcon.warn(info.steamID, 'Too close to expected end of match.');
+        return;
+      }
+
+      // check if enough time has passed since the last vote
+      if (timeLastVote - new Date() < options.pastVoteTimer) {
+        server.rcon.warn(info.steamID, 'Not enough time has passed since the last vote.');
+        return;
+      }
+
+      // Actual vote
+      voteActive = true;
+      votePos = 1;
+      playerVotes[info.steamID] = info.message;
+      voteNeg = 0;
+      // Set reminders
+      intervalReminderBroadcasts = setInterval(() => {
+        server.broadcast(
+          'A vote to skip the current map has started. Please vote in favour with + or against with -.'
+        );
+        server.broadcast('Current counter is: ' + votePos + ' in favour, ' + voteNeg + ' against.');
```

Comment:
```
      intervalReminderBroadcasts = setInterval(async () => {
        await server.broadcast(
          'A vote to skip the current map has started. Please vote in favour with + or against with -.'
        );
        await server.broadcast('Current counter is: ' + votePos + ' in favour, ' + voteNeg + ' against.');
```

We want to ensure they're done in the correct order so we should `await` the server to confirm that the broadcast has been received/made first before making the second.